### ISSUE TITLE
epochSize vs minibatchSize vs randomizationWindow 

### ISSUE THREAD

**Issue by avader906** [Microsoft/CNTK#1858]
What is the use case when 
``` 
epochSize < total training corpus 
     and with minibatchSize < randomizationWindow  < epochSize 
``` 
Does the training rolls randomizationWindow over available data independently of an end of epoch - or the training restarts from the beginning of the training dataset? In other words - in this use case - does CNTK use the entire training corpus? 

NB everything in samples, CNTKTextFormatReader

**Comment by eldakms** [comment]
Yes, it uses all available data. Maybe this will clarify better. 

Imaging an infinite timeline, along it the corpus is replicated infinite times.
|--------corpus(sweep 0)---------| |--------corpus(sweep 1)---------| |--------corpus(sweep 2)---------| ...

Each sweep is randomized - firstly we randomize chunks then we create a rolling window of chunks and randomizes samples/sequences inside:
|------randomized(seed 0)-------| |------randomized(seed 1)-------| |------randomized(seed 2)-------|...

The epoch is simply moving along this global time line according to the number of samples specified (or sweeps for "auto" in brain script):
|----epoch1---------||----epoch2---------||----epoch3---------|.....

Hope this helps.

**Comment by avader906** [comment]
@eldakms thank you vm. i've just read blockrandomizer but your explanation is much more clear. maybe the doc could be updated - it is not obvious at first?

**Comment by eldakms** [comment]
Thank you for the feedback! I will update the documentation sometime this week.
